tests/test_process.py

```diff
@@ -1,67 +1,242 @@
+        "final_format": "json",
+        "provenance": ["syn1111111"],
+        "destination": "syn1111113",
+        "custom_transformations": "test_transformation",
+    }
+}
+
+
+def test_process_dataset_custom_transformations():
+    with patch.object(
+        extract, "get_entity_as_df", return_value=pd.DataFrame
+    ), patch.object(
+        transform, "standardize_column_names", return_value=pd.DataFrame
+    ), patch.object(
+        transform, "standardize_values", return_value=pd.DataFrame
+    ), patch.object(
+        transform, "apply_custom_transformations", return_value=pd.DataFrame
+    ) as patch_custom_transform, patch.object(
+        load, "df_to_json", return_value="path/to/json"
+    ), patch.object(
+        load, "load", return_value=None
+    ):
+        process.process_dataset(dataset_obj=dataset_object_custom_transform)
+        patch_custom_transform.assert_called_once_with(
+            datasets={"test_file_1": pd.DataFrame},
+            dataset_name="neuropath_corr",
+            dataset_obj={
+                "files": [{"name": "test_file_1", "id": "syn1111111", "format": "csv"}],
+                "final_format": "json",
+                "provenance": ["syn1111111"],
+                "destination": "syn1111113",
+                "custom_transformations": "test_transformation",
+            },
+        )
+
+
+dataset_object_agora_rename = {
+    "neuropath_corr": {
+        "files": [{"name": "test_file_1", "id": "syn1111111", "format": "csv"}],
+        "final_format": "json",
+        "provenance": ["syn1111111"],
+        "destination": "syn1111113",
+        "agora_rename": {"col_1": "new_col_1", "col_2": "new_col_2"},
+    }
+}
+
+
+def test_process_dataset_with_agora_rename():
+    with patch.object(
+        extract, "get_entity_as_df", return_value=pd.DataFrame
+    ), patch.object(
+        transform, "standardize_column_names", return_value=pd.DataFrame
+    ), patch.object(
+        transform, "standardize_values", return_value=pd.DataFrame
+    ), patch.object(
+        transform, "rename_columns", return_value=pd.DataFrame
+    ) as patch_rename_columns, patch.object(
+        load, "df_to_json", return_value="path/to/json"
+    ), patch.object(
+        load, "load", return_value=None
+    ):
+        process.process_dataset(dataset_obj=dataset_object_col_rename)
+        patch_rename_columns.assert_called_once_with(
+            df=pd.DataFrame, column_map={"col_1": "new_col_1", "col_2": "new_col_2"}
+        )
+
+
+def test_process_dataset_type_dict():
+    with patch.object(
+        extract, "get_entity_as_df", return_value=pd.DataFrame
+    ), patch.object(
+        transform, "standardize_column_names", return_value=pd.DataFrame
+    ), patch.object(
+        transform,
+        "standardize_values",
+        return_value=dict(),  # test if it is a dictionary later - not sure if this is appropriate way to do it
+    ), patch.object(
+        load, "dict_to_json", return_value="path/to/json"
```

Comment:
What you can do is patch all those functions in the `setup_method` in the class and do:

```
patch_other_func.assert_not_called()
```